examples/dreamer/dreamer.py

```diff
@@ -0,0 +1,355 @@
+        #     for device in cfg.collector_devices
+        # ],
+    )
+
+    replay_buffer = make_replay_buffer(device, cfg)
+
+    recorder = transformed_env_constructor(
+        cfg,
+        video_tag=video_tag,
+        norm_obs_only=True,
+        stats=stats,
+        logger=logger,
+    )()
+
+    # remove video recorder from recorder to have matching state_dict keys
+    if cfg.record_video:
+        recorder_rm = TransformedEnv(recorder.base_env)
+        for transform in recorder.transform:
+            if not isinstance(transform, VideoRecorder):
+                recorder_rm.append_transform(transform)
+    else:
+        recorder_rm = recorder
+
+    if isinstance(create_env_fn, ParallelEnv):
+        recorder_rm.load_state_dict(create_env_fn.state_dict()["worker0"])
+        create_env_fn.close()
+    elif isinstance(create_env_fn, EnvCreator):
+        recorder_rm.load_state_dict(create_env_fn().state_dict())
+    else:
+        recorder_rm.load_state_dict(create_env_fn.state_dict())
+    # reset reward scaling
+    for t in recorder.transform:
+        if isinstance(t, RewardScaling):
+            t.scale.fill_(1.0)
+            t.loc.fill_(0.0)
+
+    record = Recorder(
+        record_frames=cfg.record_frames,
+        frame_skip=cfg.frame_skip,
+        policy_exploration=policy,
+        recorder=recorder,
+        record_interval=cfg.record_interval,
+        log_keys=cfg.recorder_log_keys,
+    )
+
+    final_seed = collector.set_seed(cfg.seed)
+    print(f"init seed: {cfg.seed}, final seed: {final_seed}")
+    ## Training loop
+    collected_frames = 0
+    pbar = tqdm.tqdm(total=cfg.total_frames)
+    r0 = None
+    path = Path("./log")
+    path.mkdir(exist_ok=True)
+
+    scaler = GradScaler()
+    torch.cuda.empty_cache()
+    for i, tensordict in enumerate(collector):
+
+        # update weights of the inference policy
+        collector.update_policy_weights_()
+
+        if r0 is None:
+            r0 = tensordict["reward"].mean().item()
+        pbar.update(tensordict.numel())
+        if "mask" in tensordict.keys():
+            current_frames = tensordict["mask"].sum()
+            tensordict = tensordict[tensordict.get("mask").squeeze(-1)]
+        else:
+            current_frames = tensordict.numel()
+            collected_frames += current_frames
+        tensordict = tensordict.reshape(-1, cfg.batch_length)
+        replay_buffer.extend(tensordict.cpu())
+
+        if collected_frames >= cfg.init_random_frames:
+            for j in range(cfg.optim_steps_per_batch):
+                # sample from replay buffer
+                sampled_tensordict = replay_buffer.sample(cfg.batch_size).to(device)
+
+                with autocast(dtype=torch.float16):
+                    model_loss_td, sampled_tensordict = world_model_loss(
+                        sampled_tensordict
+                    )
+                    if cfg.record_video:
+                        world_model_td = sampled_tensordict.clone().select(
```

Comment:
careful here: you clone the whole thing then select, I would do the opposite

Reply:
Was doing the opposite before but this was changing the original tensordict, a lot of keys were missing for the actor part after, so that's why i reverted it.

Reply:
no select does not change the original tensordict! Or there is a serious bug!